Main.java

```diff
@@ -1,90 +1,192 @@
 import entities.Cat;
 import entities.Dog;
 import entities.Horse;
-import pattern.Animal;
 import pattern.Color;
 
+import java.util.ArrayList;
+import java.util.InputMismatchException;
+import java.util.List;
 import java.util.Scanner;
+import java.util.regex.Pattern;
 
 public class Main {
 
-    public static void main(String[] args) {
-
+    public static void main(String[] args) throws InvalidAnimalName {
```

Comment:
Let`s leave only main method at Main class.
All other methods should be moved to the new class